src/core/net/socket.hpp

```diff
@@ -174,6 +177,22 @@ class MessageInfo : public otMessageInfo, public Clearable<MessageInfo>
      */
     void SetLinkInfo(const void *aLinkInfo) { mLinkInfo = aLinkInfo; }
 
+    /**
+     * This method returns a pointer to the Link Info as a `ThreadLinkInfo`.
+     *
+     * @returns A pointer to Thread Link Info.
+     *
+     */
+    const ThreadLinkInfo *GetThreadLinkInfo(void) const { return reinterpret_cast<const ThreadLinkInfo *>(mLinkInfo); }
```

Comment:
does static_cast works?

Reply:
yes, I think it would work. In this case, since we are casting from ` void *` we can use `static_cast` as well. But we generally use `static_cast` for conversion between parent/sub-classes. Here we are really casting  a `void *` as a new pointer type, so I felt `reinterpret_cast` would be indicate that better.

Reply:
I tend to use `reinterpret_cast` whenever applicable.